### PR TITLE
have flutter analyze support --preview-dart-2

### PR DESCRIPTION
**Pull request by devoncarew**
- support `--preview-dart-2` in both `flutter analyze` and `flutter analyze --watch`
- add both positive and negative tests for each

Not yet ready for review - requires a roll to a post `2.0.0-dev.22.0` sdk

### CODE REVIEW

**Comment by devoncarew** [comment]
To make progress on this PR, we'll need https://github.com/flutter/flutter/pull/14670 to land.

**Review by bwilkerson — APPROVED** [review]

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -57,9 +58,15 @@ class AnalyzeCommand extends FlutterCommand {
...
   Future<Null> runCommand() {
     if (argResults['watch']) {
-      return new AnalyzeContinuously(argResults, runner.getRepoPackages()).analyze();
+      return new AnalyzeContinuously(
+        argResults, runner.getRepoPackages(), previewDart2: argResults['preview-dart-2']
+      ).analyze();
     } else {
-      return new AnalyzeOnce(argResults, runner.getRepoPackages(), workingDirectory: workingDirectory).analyze();
+      return new AnalyzeOnce(
+        argResults, runner.getRepoPackages(),
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Nit: can you put these on a separate line

**Review by pq — APPROVED** [review]
👍

File: packages/flutter_tools/lib/src/commands/analyze_once.dart [hunk 2]
```diff
@@ -23,13 +23,18 @@ typedef bool FileFilter(FileSystemEntity entity);
 
 /// An aspect of the [AnalyzeCommand] to perform once time analysis.
 class AnalyzeOnce extends AnalyzeBase {
-  AnalyzeOnce(ArgResults argResults, this.repoPackages, { this.workingDirectory }) : super(argResults);
+  AnalyzeOnce(ArgResults argResults, this.repoPackages, {
+    this.workingDirectory,
+    this.previewDart2,
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/analyze_once.dart]: Default to false?

**Review by tvolkert — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/commands/analyze_once.dart (hunk 2 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/commands/analyze_once.dart]: ah, yup

File: packages/flutter_tools/lib/src/commands/analyze.dart (hunk 1 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/commands/analyze.dart]: 👍

**Review by devoncarew — COMMENTED** [review]
Thanks! updated

**Pull request merged**